master/internal/api_workspace.go

```diff
@@ -119,6 +119,13 @@ func (a *apiServer) getWorkspaceAndCheckCanDoActions(
 	return w, *curUser, nil
 }
 
+func (a *apiServer) workspaceHasModels(ctx context.Context, workspaceID int32) (bool, error) {
```

Comment:
This change from a users perspective feels a little strange to me.

So deleting a workspace will automatically delete experiments, templates, and NTSC, but not models? Maybe this is something we need product input on, but what do you think about being more consistent and just deleting models attached to the workspace?

Reply:
I also wondered about that; I assumed it was because models were considered "high value". I'll check with product on it.

Reply:
Heard back from product here: https://hpe-aiatscale.slack.com/archives/C06GMG83ZE0/p1707234970325459?thread_ts=1707234524.249439&cid=C06GMG83ZE0

**TL;DR** for now, do not delete models along with a workspace as models are one of the most valuable artifacts generated by Determined/MLDE